python/graphstorm/model/node_glem.py

```diff
@@ -0,0 +1,311 @@
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    GLEM model for node prediction task in GraphStorm.
+"""
+import os
+from copy import deepcopy
+import torch as th
+import dgl
+
+from .gnn import GSOptimizer
+from .node_gnn import GSgnnNodeModel, GSgnnNodeModelBase
+
+class GLEM(GSgnnNodeModelBase):
+    """
+    GLEM model (https://arxiv.org/abs/2210.14709) for node-level tasks.
+    Parameters
+    ----------
+    alpha_l2norm: float
+        Coef for l2 norm of unused weights
+    em_order_gnn_first: bool
+        In the EM training, set true to train GNN first, train LM first otherwise
+    inference_using_gnn: bool
+        Set true to use GNN in inference time, otherwise inference with LM
+    pl_weight: float
+        Weight for the pseudo-likelihood loss in GLEM's loss function.
+    """
+    def __init__(self,
+                 alpha_l2norm,
+                 em_order_gnn_first=True,
+                 inference_using_gnn=True,
+                 pl_weight=0.5
+                 ):
+        super(GLEM, self).__init__()
+        self.alpha_l2norm = alpha_l2norm
+        self.em_order_gnn_first = em_order_gnn_first
+        self.inference_using_gnn = inference_using_gnn
+        self.pl_weight = pl_weight
+        self.lm = GSgnnNodeModel(alpha_l2norm)
+        self.gnn = GSgnnNodeModel(alpha_l2norm)
+        self.training_lm = not em_order_gnn_first
+
+    def init_optimizer(self, lr, sparse_optimizer_lr, weight_decay, lm_lr=None):
+        """Initialize optimzer, which will be stored in self.lm._optimizer, self.gnn._optimizer
+        """
+        sparse_params = self.gnn.get_sparse_params()
+        if len(sparse_params) > 0:
+            emb_optimizer = dgl.distributed.optim.SparseAdam(sparse_params, lr=sparse_optimizer_lr)
+            sparse_opts = [emb_optimizer]
+        else:
+            sparse_opts = []
+
+        dense_params = self.gnn.get_dense_params() + self.lm.get_dense_params()
+        if len(dense_params) > 0:
+            optimizer = th.optim.Adam(dense_params, lr=lr,
+                                      weight_decay=weight_decay)
+            dense_opts = [optimizer]
+        else:
+            dense_opts = []
+
+        lm_params = self.lm.get_lm_params()
+        if len(lm_params) > 0:
+            lm_optimizer = th.optim.Adam(lm_params, \
+                                         lr=lm_lr if lm_lr is not None else lr,
+                                         weight_decay=weight_decay)
+            lm_opts = [lm_optimizer]
+        else:
+            lm_opts = []
+        self._optimizer = GSOptimizer(dense_opts=dense_opts,
+                                      lm_opts=lm_opts,
+                                      sparse_opts=sparse_opts)
+
+    def create_optimizer(self):
```

Comment:
is the code covered by lint? it requires a docstring.

Reply:
Not covered by lint somehow, will add.

Reply:
can you add lint check?

Reply:
I tried to play around adding it but it conflicts with `@abstractmethod` and its docstring defined in parent class. This seems to be a known issue in pylint: https://github.com/pylint-dev/pylint/issues/4916